iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm

```diff
@@ -667,42 +660,75 @@ - (void)updateBookmark:(MWMMarkID)bookmarkId
     bookmark->SetCustomName(title.UTF8String);
 }
 
-- (void)moveBookmark:(MWMMarkID)bookmarkId
-           toGroupId:(MWMMarkGroupID)groupId {
-    auto const currentGroupId = self.bm.GetBookmark(bookmarkId)->GetGroupId();
-    auto editSession = self.bm.GetEditSession();
-    if (groupId != kml::kInvalidMarkGroupId)
-      editSession.MoveBookmark(bookmarkId, currentGroupId, groupId);
+- (void)updateBookmark:(MWMMarkID)bookmarkId setColor:(MWMBookmarkColor)color {
+  auto editSession = self.bm.GetEditSession();
+
+  auto bookmark = editSession.GetBookmarkForEdit(bookmarkId);
+  if (!bookmark) {
+    NSAssert(false, @"Invalid bookmark id: @%llu", bookmarkId);
+    return;
+  }
+
+
+  auto kmlColor = kmlColorFromBookmarkColor(color);
+  if (kmlColor != bookmark->GetColor())
+    self.bm.SetLastEditedBmColor(kmlColor);
+
+  bookmark->SetColor(kmlColor);
+}
+
+- (void)moveBookmark:(MWMMarkID)bookmarkId toGroupId:(MWMMarkGroupID)groupId {
+  auto const currentGroupId = self.bm.GetBookmark(bookmarkId)->GetGroupId();
+  auto editSession = self.bm.GetEditSession();
+  if (groupId != kml::kInvalidMarkGroupId)
+    editSession.MoveBookmark(bookmarkId, currentGroupId, groupId);
 }
 
 - (void)updateTrack:(MWMTrackID)trackId
```

Comment:
Does it make sense to introduce separate methods to move track, to rename it, and to change its color?

Reply:
For now, we have 3 edit cases:
- only change the color (from color icon)
- only change the group (from leading swipe action)
- change the color, the group and name from Edit bookmark screen by tapping `Save button`

So these three methods reflect all these actions.

It is not convenient to call
```
- (void)updateTrack:(MWMTrackID)trackId
         setGroupId:(MWMMarkGroupID)groupId
              color:(UIColor *)color
              title:(NSString *)title { 
```
when you want to change only one param.

And not convenient to call 3 methods when you want to update all at once by tapping Save.